cayenne-server/src/main/java/org/apache/cayenne/access/ObjectStore.java

```diff
@@ -201,7 +205,7 @@ synchronized ObjectDiff registerDiff(Object nodeId, NodeDiff diff) {
                 }
             }
 
-            objectDiff = new ObjectDiff(object);
+            objectDiff = new ObjectDiff(persistentWrapper);
```

Comment:
Was there a particular reason that ObjectDiff was modified ?
It doesn't seem to require any functionality from ObjectStorePersistentWrapper ?

Reply:
Yes, though it's not obviuos. Diff must hold a reference to the ObjectStore internal state object, as otherwise it may be deleted by the GC from the Weak/Soft reference map.